### DISCUSSION TITLE
Cant set Budget for revenue/Income transaction. Am I miss something?

### DISCUSSION THREAD

**Discussion by PiotrEsse** [firefly-iii/firefly-iii#7041] [Firefly III]
Hi, 

> Summarize:
> 1. automatic rule dont work with assigning budget to income transactions
> 2. theres no way to assign specific budget to income/revenue transaction.

I am trying to manage one of my budget where I have expenses but also income. I can set expense transactions to specific budget but cant have option to do this for income transaction. I would like to achieve  what is the most logical think in my opinion: budgeting. But now I only have expenses but not being balanced by income - is this how budgets was meant to work?

In automate rules You can assign budget to  income transaction:
![obraz](https://user-images.githubusercontent.com/42764623/219305247-f631e674-f505-441f-92fc-0b73553c03f9.png)

But when You try to edit transaction you dont have this option. 
![obraz](https://user-images.githubusercontent.com/42764623/219305975-e25fdae7-c71f-481e-85c4-40220876a901.png)

Please help.

**Comment by JC5** [comment]
https://docs.firefly-iii.org/firefly-iii/concepts/budgets/

**Reply by PiotrEsse** [reply]
Thank You, but consider enebling this option for revenue/income transations. IMHO this woulr extend functionality a lot and theres all options already present. 

Additionally if You dont find this good idea - leaving option in rules to set budget isn't misleading? Just my 2Cents. BTW - Firefly is great - thank You very much for all Your trouble!

**Reply by JC5** [reply]
> leaving option in rules to set budget isn't misleading?

I can't tell from a rule if a transaction is going to be a deposit unfortunately.

**Reply by PiotrEsse** [reply]
Thak You!

**Reply by johnydroptables** [reply]
Broken link?
I'm also looking for this function - my typical example is an investment property, it has revenue and expenses, a budget seems like a good way to view that.

**Reply by dberrocal-git** [reply]
I have been using the part of the budget to make a renovation at home, I have left over material, and I have been refunded some money for it, how can I reflect the income in the budget to refine/fine-tune it?

Is it better to use another feature for this kind of things?

**Reply by mystuffspt** [reply]
> > leaving option in rules to set budget isn't misleading?
> 
> I can't tell from a rule if a transaction is going to be a deposit unfortunately.

https://docs.firefly-iii.org/references/firefly-iii/rule-triggers/#properties

> Rule triggers
> 1. Type. Accepts English terms only: "withdrawal", "deposit" or "transfer"

Now this is possible, right?

I would like to split my deposit into budgets: xx€ for Budget A, xx% for Budget B and the remaining for Budget X.

**Reply by JC5** [reply]
You could use the API or webhooks to do that. But you will have to develop something yourself. 

What you propose is something that fits more in zero based budgeting.

**Reply by mystuffspt** [reply]
Fair enough. I will see the docs regarding API usage.

Regarding budgeting, I see the budgets as "what money do I have and what for", and they are adjusted according the income. So, every cent that I own, I know where I will use it for.
